warehouse/query-core/src/main/java/datawave/query/predicate/ChainableEventDataQueryFilter.java

```diff
@@ -0,0 +1,231 @@
+package datawave.query.predicate;
+
+import datawave.query.attributes.Document;
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+
+import javax.annotation.Nullable;
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * Wrap multiple EventDataQueryFilter objects and in order process them
+ */
+public class ChainableEventDataQueryFilter implements EventDataQueryFilter {
+    private List<EventDataQueryFilter> filters;
+    
+    public ChainableEventDataQueryFilter() {
+        filters = new ArrayList<>();
+    }
+    
+    private ChainableEventDataQueryFilter(ChainableEventDataQueryFilter other) {
+        // create a copy of all the filters
+        this.filters = new ArrayList<>(other.filters);
+    }
+    
+    public void addFilter(EventDataQueryFilter filter) {
+        filters.add(filter);
+    }
+    
+    /**
+     * Call startNewDocument on all filters in order
+     * 
+     * @param documentKey
+     */
+    @Override
+    public void startNewDocument(Key documentKey) {
+        for (int i = 0; i < filters.size(); i++) {
+            filters.get(i).startNewDocument(documentKey);
+        }
+    }
+    
+    /**
+     * Call apply on filters in FIFO order, until one is false otherwise return true
+     * 
+     * @param entry
+     * @return
+     */
+    @Override
+    public boolean apply(@Nullable Map.Entry<Key,String> entry) {
+        Iterator<EventDataQueryFilter> iterator = filters.iterator();
+        boolean result = true;
+        
+        while (iterator.hasNext() && result) {
+            result = iterator.next().apply(entry);
+        }
+        
+        return result;
+    }
+    
+    /**
+     * Call keep on filters in FIFO order, until one is false, otherwise return true
+     * 
+     * @param k
+     * @return
+     */
+    @Override
+    public boolean keep(Key k) {
+        Iterator<EventDataQueryFilter> iterator = filters.iterator();
+        boolean result = true;
+        
+        while (iterator.hasNext() && result) {
+            result = iterator.next().keep(k);
+        }
+        
+        return result;
+    }
+    
+    /**
+     * Get the maximum start key from all filters
```

Comment:
hmm, are we sure we want the maximum?

Reply:
I assumed we'd want the most restrictive of the filter stack for all methods. Do you think this is the wrong assumption?